### ISSUE TITLE
Unable to preventDefault inside passive event listener invocation.

### ISSUE THREAD

**Issue by itpika** [xtermjs/xterm.js#3110]
## Details
- Browser and browser version:  chrome: 版本 84.0.4147.89
- OS version: windows
- xterm.js version: 4.9.0

### Steps to reproduce
1. After initializing the window（new Terminal(...)）
2. When I click inside the command line window, the Chrome console displays the following error. The window cannot get the cursor and cannot be entered
![image](https://user-images.githubusercontent.com/29301013/94237314-4099cd80-ff41-11ea-9050-126b20a99ad1.png)

**Comment by jerch** [comment]
@itpika This needs abit more debug info:
- which event do you try to prevent (mousedown?)
- exact line of the code the exception happens

Ideally create a repro, that can be replayed by others.

**Comment by Tyriar** [comment]
Yeah I think we need a better repro for this, using a break on the exception in the demo would be useful. It doesn't look like mouse events in `onMouseDown` as called out in the stack should be passive listeners after a look through the code.

**Comment by SmarterInChild** [comment]
I also have this problem,so I prevent ['click', 'dblclick', 'mousedown', 'mouseup'] events。but terminal still lost cursor.
but if I use right click,the cursor recovers

**Comment by SmarterInChild** [comment]
I fix this problem, but console still displays errors
`                    document.getElementById('terminal').addEventListener('click', function(e) {
                        term.focus();
                    });
`

**Comment by itpika** [comment]
> Yeah I think we need a better repro for this, using a break on the exception in the demo would be useful. It doesn't look like mouse events in `onMouseDown` as called out in the stack should be passive listeners after a look through the code.

hello

I a recent project in the xterm, https://github.com/itpika/knight.

This is a Redis visualization tool developed using electron. I considered the compatibility problem of electron or VUE before, but it was ruled out later.

If you have time to clone my project and run it, you can reproduce this problem. Thank you very much.
![微信图片_20201113152336](https://user-images.githubusercontent.com/29301013/99040594-5f712380-25c4-11eb-9c85-8b160e39e783.png)
![微信图片_2](https://user-images.githubusercontent.com/29301013/99040614-65ff9b00-25c4-11eb-98c4-48a18be90469.png)

**Comment by itpika** [comment]
> I fix this problem, but console still displays errors
> ` document.getElementById('terminal').addEventListener('click', function(e) { term.focus(); });`

😂I was foolish enough to think that clicking the mouse on the window would automatically get the cursor, but I actually had to register the click event manually to get the cursor.

I add the term. focus(). Fixed a problem where command line clicks failed to get the cursor. However, the browser will still prompt the preventDefault problem. I chose to ignore it because my program is working fine
